test/partialAttribute/nested-html/stamped_from_polymer_template.html

```diff
@@ -157,147 +157,140 @@
+                        expect(include.shadowRoot).to.equal(shadowRoot);
+                    });
                 });
-            });
 
-            describe('after `dom-bind`` changed the sub-partial property (just one scope) to new one with thethat results with the same merged Html', function () {
-                var changedScopeWithSameHtml;
-                beforeEach(function(done) {
-                    // IDEA: promisify juicy-html
-                    importedTemplate.addEventListener('stamping', function onceStamped(){
-                        importedTemplate.removeEventListener('stamping', onceStamped);
+                describe('after `dom-bind`` changed the sub-partial (just one scope) to new a one that results with the same merged Html', function () {
+                    var changedScopeWithSameHtml;
+                    beforeEach(function() {
+                        sinon.stub(importedTemplate, "_setPendingPropertyOrPath");
                         changedScopeWithSameHtml = {
                             'Html': 'scope1Html',
                             'doesItWork': 'still works!'
                         };
                         domBind.set('model.Page.scope1', changedScopeWithSameHtml);
-                        done();
                     });
-                });
 
-                it('should attach same `.Html` property as href attribute and propery for `<imported-template>`', function () {
-                    expect(importedTemplate.getAttribute("href")).to.equal('/sc/htmlmerger?scope1=scope1Html&scope2=scope2Html%3B%2C%2F%3F%3A%40%26%3D%2B%24&scope3=');
-                    expect(importedTemplate.href).to.equal('/sc/htmlmerger?scope1=scope1Html&scope2=scope2Html%3B%2C%2F%3F%3A%40%26%3D%2B%24&scope3=');
-                });
-                it('should attach new partial model to `imported-template` immediately', function () {
-                    expect(importedTemplate.model.scope1).to.equal(changedScopeWithSameHtml);
+                    it('should keep same `.Html` property as href attribute and propery for `<imported-template>`', function () {
+                        expect(importedTemplate.getAttribute("href")).to.equal('/sc/htmlmerger?scope1=scope1Html&scope2=scope2Html%3B%2C%2F%3F%3A%40%26%3D%2B%24&scope3=');
+                        expect(importedTemplate.href).to.equal('/sc/htmlmerger?scope1=scope1Html&scope2=scope2Html%3B%2C%2F%3F%3A%40%26%3D%2B%24&scope3=');
+                    });
+                    it('should attach new partial model to `imported-template` immediately', function () {
+                        expect(importedTemplate.model.scope1).to.equal(changedScopeWithSameHtml);
+                    });
+                    it('should forward the notification to `imported-template` immediately', function () {
+                        expect(importedTemplate._setPendingPropertyOrPath).to.be.calledWith("model.scope1", changedScopeWithSameHtml);
```

Comment:
This is not listed in the PR summary. Can you explain why it is needed? I thought that we don't need to forward the notification because `juicy-html` resets the model anyway.

Reply:
Sorry, I'm adding it to the summary.

Reply:
We do need it, as the model objectis still the same, you are changing just single App partial view -model, also the URL stays the same, the HTML is not reloaded, so there is no need to re-attach the model.

Imagine the case:
```js
viewModel = {
  scope1: {
    name: 'World',
    Html: 'hello.html'
  }
  AppB: {
    foo: 'bar',
    Html: 'baz.html'
  }
}
```
`hello.html&baz.html`:
```html
<imported-template-scope scope="scope1">
<template>
  <dom-bind><template>
    <span>Hello {{name}}</span>
  </template></dom-bind>
</template>
...
```

Then you change just `externalDomBind.set('viewModel.scope1', {name:'Marcin', Html:'hello.html'});`

So far (ekhm https://github.com/Starcounter/RebelsLounge/issues/97) we decided not to reload the HTML elements, just update the view with new data. Setting the `juicyHtml.model = viewModel` will not trigger any change as `juicyHtml.model === viewModel`. So we need to call notification explicitely.

Reply:
Ah, right! Now I get it. These test descriptions are easier to read in the WCT test report than in the source file.